### PR TITLE
- add libffi

### PR DESCRIPTION
**Pull request by SSE4**
Specify library name and version:  **libffi/3.2.1**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/libffi/all/test_package/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+class TclTestConan(ConanFile):
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake"
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def imports(self):
```
Comment (danimtb) [inline recipes/libffi/all/test_package/conanfile.py]: Are imports needed? running with `run_environment=True` should be enough, right?

Comment (SSE4) [reply recipes/libffi/all/test_package/conanfile.py]: I don't think it's needed. just a leftover from old days before we invented `run_environment`

File: recipes/libffi/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,271 @@
+from conans import ConanFile, tools, AutoToolsBuildEnvironment
+import os
+import platform
+from contextlib import contextmanager
+
+
+class LibffiConan(ConanFile):
+    name = "libffi"
+    version = "3.2.1"
```
Comment (uilianries) [inline recipes/libffi/all/conanfile.py]:
```suggestion
```

File: recipes/libffi/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+    description = "A portable, high level programming interface to various calling conventions"
+    topics = ("conan", "libffi")
```
Comment (uilianries) [inline recipes/libffi/all/conanfile.py]:
```suggestion
    topics = ("conan", "libffi", "runtime", "foreign-function-interface", "runtime-library")
```

File: recipes/libffi/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+    url = "https://github.com/bincrafters/conan-libffi"
```
Comment (uilianries) [inline recipes/libffi/all/conanfile.py]:
```suggestion
    url = "https://github.com/conan-io/conan-center-index"
```

**Pull request merged**